jwst/straylight/straylight.py

```diff
@@ -214,12 +185,139 @@ def correct_MRS(input_model, straylight_model):
 
     simage = convolve(straylight_image, Box2DKernel(25))
 
-    #print 'simage',simage[debug_row,0:50]
     # remove the straylight correction for the reference pixels
     simage[:, 1028:1032] = 0.0
     simage[:, 0:4] = 0.0
 
     output.data = output2_data - simage
-    #print 'out',output.data[debug_row,0:50]
-
     return output
+
+
+def correct_MRS_ModShepard(input_model, region_model,roi,power):
+    """
+    Short Summary
+    -------------
+    Corrects the MIRI MRS data for straylight using a Modified version of the
+    Shepard algorithm. Straylight is determined using an inverse distance weighting
+    function. The inverse distance weighting is determined by module
+    Shepard2DKernel(roi,power):
+
+    Parameter
+    ----------
+    input_model: data model object
+        science data to be corrected
+
+    regions_model: holds the pixel region mask for the correction
+                   slice = (band*100+slice#)
+                   gap = 0
+    roi: region of inflence (size of radius)
+    power: exponent of Shepard kernel
+    sci_ngroups:int
+        number of groups in input data
+
+
+    Returns
+    -------
+    output: data model object
+        straylight-subtracted science data
+
+    """
+
+    # Save some data parameterss for easy use later
+    nrows, ncols = input_model.data.shape
+    # mask is either non 0 for slices and 0 for gaps between slices
+    # this algorithm using the pixels from gaps between the slices for 
+    # correction
+    sliceMap = region_model.regions.copy()
+
+    #plt.imshow(sliceMap)
+    #plt.show()
+
+    # The regions mask has values of 0 and nonzero, values of 0 present pixel
+    # in a slice gap and non zero values are science pixels.  The straylight task uses data
+    # in-between the slices (also called slice gaps) of the MRS data to correct 
+    # the science data in the slices. 
+
+    # Create output as a copy of the input science data model
+    output = input_model.copy() # this is used in algorithm to
+    output_data = input_model.data.copy()
```

Comment:
Why is `output_data` needed as a separate array? Can't you just use `output.data`? They both contain the same array.